### ISSUE TITLE
panic: runtime error: index out of range

### ISSUE THREAD

**Issue by ding444** [cbednarski/hostess#22]
Every time I run a `hostess` command I get:

```
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x1776a0, 0xc420014130)
    /usr/local/Cellar/go/1.7.1/libexec/src/runtime/panic.go:500 +0x1a1
github.com/cbednarski/hostess.Hostlist.Less(0xc4207c0000, 0x810b, 0x9800, 0x0, 0x62c4, 0xc4203c8001)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/hostlist.go:78 +0x2ac
github.com/cbednarski/hostess.(*Hostlist).Less(0xc42053c180, 0x0, 0x62c4, 0x4001)
    <autogenerated>:2 +0x78
sort.doPivot(0x25db20, 0xc42053c180, 0x0, 0x810b, 0x5c190, 0xc42004f6f8)
    /usr/local/Cellar/go/1.7.1/libexec/src/sort/sort.go:121 +0x1be
sort.quickSort(0x25db20, 0xc42053c180, 0x0, 0x810b, 0x1f)
    /usr/local/Cellar/go/1.7.1/libexec/src/sort/sort.go:188 +0x83
sort.Sort(0x25db20, 0xc42053c180)
    /usr/local/Cellar/go/1.7.1/libexec/src/sort/sort.go:222 +0x80
github.com/cbednarski/hostess.(*Hostlist).Sort(0xc420010910)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/hostlist.go:140 +0x5a
github.com/cbednarski/hostess.LoadHostfile(0xd1641, 0x1a0d7b, 0x2, 0xc42004f818)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/hostfile.go:151 +0x116
github.com/cbednarski/hostess.AlwaysLoadHostFile(0xc4200b20e0, 0x66855)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/commands.go:53 +0x26
github.com/cbednarski/hostess.Ls(0xc4200b20e0)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/commands.go:196 +0x5b
github.com/cbednarski/hostess/vendor/github.com/codegangsta/cli.Command.Run(0x1a119a, 0x4, 0x0, 0x0, 0xc420014ce0, 0x1, 0x1, 0x1a6c32, 0x1e, 0x0, ...)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/vendor/github.com/codegangsta/cli/command.go:118 +0x7bd
github.com/cbednarski/hostess/vendor/github.com/codegangsta/cli.(*App).Run(0xc4200ac000, 0xc42000c280, 0x2, 0x2, 0x0, 0x0)
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/vendor/github.com/codegangsta/cli/app.go:158 +0x6da
main.main()
    /private/tmp/hostess-20160927-16340-1wrlq4q/hostess-0.2.1/src/github.com/cbednarski/hostess/cmd/hostess/hostess.go:117 +0x780
```

I should note that my hosts file is **very** large, with approx. 35,000 entries (I use https://github.com/StevenBlack/hosts).  My use case may not align with that of the project.

**Comment by ding444** [comment]
A workaround is using the `HOSTESS_PATH` to point to the `myhosts` file in the https://github.com/StevenBlack/hosts project.

Works great!

**Comment by cbednarski** [comment]
Thanks for the report. I'll see if I can repro using https://raw.githubusercontent.com/StevenBlack/hosts/master/hosts

**Comment by leominov** [comment]
@cbednarski, I guess its a mistake in comments parser, here is a problem block:
```
# comment: 123
0.0.0.0 hostname
```
After `ls`:
```
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0x1ec460, 0x8203b8080)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/cbednarski/hostess.Hostlist.Less(0x8203b8f00, 0x2, 0x2, 0x1, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:78 +0x559
github.com/cbednarski/hostess.(*Hostlist).Less(0x8203b29c0, 0x1, 0x0, 0x0)
	<autogenerated>:2 +0xb5
sort.insertionSort(0x88203fe578, 0x8203b29c0, 0x0, 0x2)
	/usr/local/go/src/sort/sort.go:32 +0x6f
sort.quickSort(0x88203fe578, 0x8203b29c0, 0x0, 0x2, 0x4)
	/usr/local/go/src/sort/sort.go:214 +0x202
sort.Sort(0x88203fe578, 0x8203b29c0)
	/usr/local/go/src/sort/sort.go:229 +0x74
github.com/cbednarski/hostess.(*Hostlist).Sort(0x8203bcad0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostlist.go:140 +0x70
github.com/cbednarski/hostess.LoadHostfile(0x8203bcac0, 0x0, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/hostfile.go:151 +0x10a
github.com/cbednarski/hostess.AlwaysLoadHostFile(0x8204363c0, 0xf56ca)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/commands.go:53 +0x24
github.com/cbednarski/hostess.Ls(0x8204363c0)
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/commands.go:196 +0x4e
github.com/codegangsta/cli.Command.Run(0x22eec8, 0x4, 0x0, 0x0, 0x8203b8ab0, 0x1, 0x1, 0x261c00, 0x1e, 0x0, ...)
	/Users/cbednarski/go/src/github.com/codegangsta/cli/command.go:179 +0x1399
github.com/codegangsta/cli.(*App).Run(0x82042a840, 0x8203b2240, 0x2, 0x2, 0x0, 0x0)
	/Users/cbednarski/go/src/github.com/codegangsta/cli/app.go:196 +0x137c
main.main()
	/Users/cbednarski/go/src/github.com/cbednarski/hostess/cmd/hostess/hostess.go:117 +0xa6c
```

**Comment by cbednarski** [comment]
Thanks for the report! This should be fixed along with #23 in 831904824cfb887699235007acf6dfbd70e8982b
